implementation/src/main/java/io/smallrye/jwt/auth/principal/ClaimSubPathResolver.java

```diff
@@ -0,0 +1,53 @@
+package io.smallrye.jwt.auth.principal;
```

Comment:
I'd like to avoid introducing a dedicated class for checking sub paths only, while the `groups` are checked inside the parser, lets keep it inside the parser code for the moment. Good idea to try to push it outside,  but we can attempt to generalize later so that a single code is dealing will all the claim paths, with the only custom bit being checking the final segment, but we can take care of it later

Reply:
This was because I needed the same logic to retrieve the value in the jose4j validator.